### PR TITLE
Add in complex trace type svg components

### PR DESCRIPTION
**Pull request by aulneau**
This implements the new trace type icons that I've added to the `plotly-icons` lib. 

Once this https://github.com/plotly/plotly-icons/pull/16 is merged and we have a new version on npm, I'll bump it here, test it and we should be good to go!

resolves https://github.com/plotly/react-chart-editor/issues/379 https://github.com/plotly/react-chart-editor/issues/375

### CODE REVIEW

File: src/lib/strings.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*eslint-disable */
+
+/**
```
Comment (VeraZab) [inline src/lib/strings.js]: why are we eslint disabling?

File: src/lib/index.js [hunk 2]
```diff
@@ -26,6 +27,16 @@ import {
...
 import striptags from './striptags';
+import {
+  capitalize,
+  lowerCase,
+  upperCase,
+  removeNonWord,
+  camelCase,
+  pascalCase,
+} from './strings';
+
```
Comment (VeraZab) [inline src/lib/index.js]: thanks :)

File: src/lib/strings.js (hunk 1 above)
Comment (aulneau) [reply src/lib/strings.js]: my IDE was giving me errors related to the various replace regex like strings that are used throughout these scripts:
```
.replace(/\-/g, ' ')
.replace(/(\d)(?=(\d{1})+$)/g, '$1 ')
.replace(/\s[a-z]/g, upperCase)
.replace(/\s+/g, '')
.replace(/^[A-Z]/g, lowerCase);
```

**Comment by VeraZab** [comment]
let's change plotly-icons here to `latest`: https://github.com/plotly/react-chart-editor/blob/master/package.json#L17

File: src/lib/strings.js (hunk 1 above)
Comment (VeraZab) [reply src/lib/strings.js]: ok, we can keep the lint disabling

**Comment by VeraZab** [comment]
💃 once plotly-icons version is changed! thanks @aulneau !!

**Pull request merged**